docs/en/ingest-management/commands.asciidoc

```diff
@@ -31,31 +31,88 @@ NOTE: You might need to log in as a root user to run these commands.
+start and manage the service. This command is also useful for testing
+{agent} prior to installing it.
 
-We recommend that you run this command as the root user because some
+TIP: To enroll an {agent} _and_ install it as a service, use the
+<<elastic-agent-install-command,`install` command>> instead.
+
+We recommend that you run the `enroll` command as the root user because some
 integrations require root privileges to collect sensitive data. This command
 overwrites the `elastic-agent.yml` file in the agent directory.
 
+This command includes optional flags to set up <<fleet-server,{fleet-server}>>.
+
+IMPORTANT: This command enrolls the {agent} in {fleet}; it does not start the
+agent. To start the agent, either <<start-elastic-agent,start the service>>, if
+one exists, or use the <<elastic-agent-run-command,`run` command>> to start the
+agent from a terminal.
+
+
 [discrete]
 === Synopsis
 
+//QUESTION: Which flags are relevant when running a Fleet
+//Server agent vs a regular agent? Which flags are required? (We should list
+//required flags first and they shouldn't be wrapped in square brackets). I'm
+//making some guesses here, but I'm not sure about the Ca and certificate
+//settings.
+
+//Also, should we include deprecated flags (like kibana-url) in the synopsis? I think
+//probably not.
+
+
+To enroll the {agent} in {fleet}:
+
 [source,shell]
 ----
-elastic-agent enroll --kibana-url=<kibana_url> --enrollment-token=<enrollment_token> [--ca-sha256 <string>]
-                     [--certificate-authorities <string>] [--force] [--help]
-                     [--insecure] [global-flags]
+elastic-agent enroll [--ca-sha256 <string>]
+                     [--certificate-authorities <string>]
+                     [--enrollment-token <string>]
+                     [--force]
+                     [--help]
+                     [--insecure ]
+                     [--url string]
+                     [global-flags]
+----
+
+To enroll the {agent} in {fleet} and set up {fleet-server}:
+
+[source,shell]
 ----
+elastic-agent enroll [--ca-sha256 <string>]
+                     [--certificate-authorities <string>]
+                     [--fleet-server-cert <string>] <1>
+                     [--fleet-server-cert-key <string>]
+                     [--fleet-server-es <string>]
+                     [--fleet-server-es-ca <string>]
+                     [--fleet-server-host <string>]
+                     [--fleet-server-insecure-http]
+                     [--fleet-server-policy <string>]
+                     [--fleet-server-port <uint16>]
+                     [--fleet-server-service-token <string>]
+                     [--force]
+                     [--help]
+                     [--insecure ]
+                     [global-flags]
+----
+<1> If no `fleet-server-cert*` flags are specified, {agent} auto-generates a
+self-signed certificate with the hostname of the machine.
+
+//QUESTION: Does this mean that the agent enrolling in this Fleet Server needs
```

Comment:
If Fleet Server is employing the self-signed certs, then yes, any Agents that enroll to it will need the --insecure flag.  I have a task on my list to add this to the trouble-shooting docs, unless you want to call it out here?

And yes, using the self-signed certs temporarily is less secure, but all communication between Agents and a Fleet server will be encrypted regardless, and we should mention that.

Reply:
Just for clarity on the question the `--insecure` flag is only required on a remote Elastic Agent enrolling into this Elastic Agent. The Elastic Agent that is being bootstrapped (aka. itself) does not need `--insecure`. The bootstrapping Elastic Agent is using a 100% verified certificate chain, as it has its own CA that is used to verify the certificate.